src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +47,24 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(e) => handleChangeQuery(e.target.value)}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            <button
+              aria-label="remove"
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => handleChangeQuery()}
```

Comment:
not fixed 
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/70009205/f9cd956f-35c8-4b72-8e03-398523f35a48)